src/gei/Commands/MigrateArchiveRepoCommand.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.Threading.Tasks;
+
+namespace OctoshiftCLI.GithubEnterpriseImporter.Commands
+{
+    public class MigrateArchiveRepoCommand : Command
+    {
+        private readonly OctoLogger _log;
+        private readonly ISourceGithubApiFactory _sourceGithubApiFactory;
+        private readonly ITargetGithubApiFactory _targetGithubApiFactory;
+        private readonly IAzureApiFactory _azureApiFactory;
+        private readonly EnvironmentVariableProvider _environmentVariableProvider;
+        private readonly MigrateRepoCommand _migrateRepoCommand;
+
+        private const int ARCHIVE_GENERATION_TIMEOUT_IN_HOURS = 10;
+        private const int CHECK_STATUS_DELAY_IN_MILLISECONDS = 10000; // 10 seconds
+        private const string GIT_ARCHIVE_FILE_NAME = "gitArchive.tar.gz";
+        private const string METADATA_ARCHIVE_FILE_NAME = "metadataArchive.tar.gz";
+
+        public MigrateArchiveRepoCommand(OctoLogger log, ISourceGithubApiFactory sourceGithubApiFactory, ITargetGithubApiFactory targetGithubApiFactory, EnvironmentVariableProvider environmentVariableProvider, IAzureApiFactory azureApiFactory, MigrateRepoCommand migrateRepoCommand) : base("migrate-archive-repo")
```

Comment:
@dylan-smith Arin said we should mention that this can totally be put into the migrate repo command as a flag (like you were mentioning on the call earlier) rather than a separate command, but for now we kept it separate to not complicate things. Merging them can be in a followup 👍